src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -524,6 +557,34 @@ internal static OperationStatus EncodeUtf8Shim(TextEncoder encoder, ReadOnlySpan
             return encoder.EncodeUtf8(utf8Source, utf8Destination, out bytesConsumed, out bytesWritten, isFinalBlock);
         }
 
+        /// <summary>
+        /// Encodes the supplied characters.
+        /// </summary>
+        /// <param name="source">A source buffer containing the characters text to encode.</param>
+        /// <param name="destination">The destination buffer to which the encoded form of <paramref name="source"/>
+        /// will be written.</param>
+        /// <param name="charsConsumed">The number of characters consumed from the <paramref name="source"/> buffer.</param>
+        /// <param name="charsWritten">The number of characters written to the <paramref name="destination"/> buffer.</param>
+        /// <param name="isFinalBlock"><see langword="true"/> if there is further source data that needs to be encoded;
+        /// <see langword="false"/> if there is no further source data that needs to be encoded.</param>
+        /// <returns>An <see cref="OperationStatus"/> describing the result of the encoding operation.</returns>
+        /// <remarks>The buffers <paramref name="source"/> and <paramref name="destination"/> must not overlap.</remarks>
+        public virtual OperationStatus Encode(
```

Comment:
Will this method ever throw? Doesn't look like it, but wanted to verify.

Reply:
It does not throw.

Reply:
Destination is a `Span<char>`. How can it be null?

Reply:
You can assign null to a span -- it just gets treated as empty. And then when the pointer is obtained, an [actual null pointer is returned](https://github.com/dotnet/corefx/blob/e30ab6c16dbd37098dc3083712fe75bf5c8c86cc/src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs#L157). Then the other code (not modified here) checks for null and throws ArgumentNullException.

However, I think we should just return OperationStatus.DestinationTooSmall instead of throwing.

Reply:
Oh... It's going through the implicit cast from array to span. Generally, structs can't be assigned null  (only default), but the hidden cast here is why it works.
```C#
Span<char> span = null; // Equivalent to Span<char> span = (char[])null;
```